main.go

```diff
@@ -4,37 +4,67 @@ import (
 	"flag"
 	"log"
 	"path/filepath"
+	"strings"
 
 	"github.com/JackTimothy/pubdoc/configuration"
 	"github.com/JackTimothy/pubdoc/confluence"
+	"github.com/JackTimothy/pubdoc/parser"
 	"github.com/joho/godotenv"
 )
 
 func main() {
-	var markdownFilePath, title string
-	flag.StringVar(&markdownFilePath, "markdownFilePath", "", "The path to the Markdown file you want to publish.")
-	flag.Parse()
-
-	title = filepath.Base(markdownFilePath)
-	title, err := confluence.FormatBase(title)
-	if err != nil {
-		log.Fatalf("Error formatting title: %v.", err)
+	var markdownFilePath, title, htmlPageContent string
+	var config configuration.Configuration
+	var err error
+
+	// parse flags
+	{
+		flag.StringVar(&markdownFilePath, "markdownFilePath", "", "The path to the Markdown file you want to publish.")
+		flag.Parse()
 	}
 
-	if err := godotenv.Load(); err != nil {
-		log.Fatalf("Error loading .env file: %v.", err)
+	// parse title
+	{
+		title = filepath.Base(markdownFilePath)
+		title, err = confluence.FormatBase(title)
+		if err != nil {
+			log.Fatalf("Error formatting title: %v.", err)
+		}
 	}
```

Comment:
I quite like this style of coding since you isolate the "point" of the code without necessarily making a function. 
In theory you can fold it away when you are convinced its benign. I like it.

Reply:
It looks like it's creating a new scope, meaning you have to declare all your variables at the beginning like an old C program. I feel like if your function is so long that you have to collapse different parts of it, it's time to break it up

Reply:
It does create inner scopes, and I think that is what is great about it. 
I don't think it would be particularly valuable to function extract this stuff since it's only ever going to be done once. 
It does read easier in my opinion due to the reasons provided above. 

I want to fight you on this one, but realistically I'm not as opinionated as you and I'll give up. 

I do really like this style though and wish more people did this. It reads so clean to me. Perhaps that's just me. 

I'm just imagining trying to work on this without ever reading it. You would read "`parse title`" then be like I don't care about the title parsing at all -> fold the code. Now it's out of the way.